rdr_service/resource/generators/_base.py

```diff
@@ -66,6 +70,23 @@ def _load_data(self, schema, data):
                     data[name] = val if type(meta) == fields.DateTime else val.date()
                 except (ParserError, TypeError):
                     pass
+            # Only provide enum name string, not full class + name.
+            if name in data and type(meta) == fields.EnumString:
+                try:
+                    data[name] = data[name].name
+                except AttributeError:
+                    pass
+
+        # Remove any extra fields that are not declared in the schema.
+        schema_fields = schema._declared_fields.keys()
+        data_fields = data.keys()
+        remove = list()
+        for field in data_fields:
+            if field not in schema_fields:
+                remove.append(field)
```

Comment:
Is there a reason you have to build a remove list and loop through it again to pop the fields, instead of doing the data.pop(field) as you find them in the first loop?  (Not a functional issue and not a big deal for small lists, just wondering)

Reply:
Removing records from the same list your iterating is a bad idea and in this specific case throws an error.